exporter/awsprometheusremotewriteexporter/auth.go

```diff
@@ -135,3 +147,25 @@ func cloneRequest(r *http.Request) *http.Request {
 	}
 	return r2
 }
+
+// FormatHeaderValues will add the name/version pair to the User-Agent request
+// header. If the extra parameters are provided they will be added as metadata to the
+// name/version pair resulting in the following format.
+// "name/version (extra0; extra1; ...)"
+// The user agent part will be concatenated with this current request's User-Agent string.
+func FormatHeaderValues(req *http.Request, name, version string, extra ...string) {
+	ua := fmt.Sprintf("%s/%s", name, version)
+	if len(extra) > 0 {
+		ua += fmt.Sprintf(" (%s)", strings.Join(extra, "; "))
+	}
+	AddToUserAgent(req, ua)
```

Comment:
You can move the AddToUserAgent body here:

curUA := req.Header.Get("User-Agent")
	if len(curUA) > 0 {
		s = curUA + " " + s
	}
	req.Header.Set("User-Agent", s)

Reply:
@rakyll I will make this change. So should I combine the 2 functions and name it as addUserAgent?